src/Repositories/RedisRepository.php

```diff
@@ -77,32 +77,60 @@ public function partitionsWithCount($queue)
         return $partitions;
     }
 
-    public function jobs($partition)
+    public function totalJobsCount($queues)
     {
         $prefix = $this->getPrefix();
         $redis = $this->getConnection();
-        $per_page = request('limit', 100);
 
-        $jobs = $redis->lrange($partition, 0, $per_page + 1);
+        $jobs_count = 0;
+
+        foreach($queues as $queue) {
+            foreach($this->partitions($queue) as $partition) {
+                $jobs_count += $redis->llen("{$prefix}:{$queue}:{$partition}");
+            }
+        }
+        return $jobs_count;
+    }
+
+    public function jobs($partition)
+    {
+        $redis = $this->getConnection();
+        $per_page = request('limit', 25);
+        $starting_at = request('starting_at', 0);
+
+        $jobs = $redis->lrange($partition, $starting_at, $per_page + $starting_at);
+        $jobs_total_pages = ceil(count($redis->lrange($partition, 0, -1)) / $per_page);
 
         $jobs_array = [];
 
-        foreach($jobs as $job)
+        foreach($jobs as $index=>$job)
         {
-            $jobs_array[] = get_class(unserialize($job));
+            $jobs_array[$index]['id'] = ($index + $starting_at);
+            $jobs_array[$index]['name'] = get_class(unserialize($job));
         }
 
-        return $jobs_array;
+        $has_more = count($jobs) > $per_page;
+
+        if($has_more) {
+            array_pop($jobs_array);
+        }
+
+        return [
+            'jobs'    => $jobs_array,
+            'has_more' => $has_more,
+            'total' => $jobs_total_pages
+        ];
     }
 
     public function job($partition, $index)
     {
-        $prefix = $this->getPrefix();
         $redis = $this->getConnection();
 
         $jobs = $redis->lrange($partition, $index, $index);
```

Comment:
here also we should generate the redis key not expecting the parameter to be a redis key.

Reply:
I know your concerns but what should we sent instead of partition name. Because it doesn't have something like id or ...
We can send index but it's not reliable I think.
What do you suggest?

Reply:
we should send queue + partition + index and we generate the key ourselves